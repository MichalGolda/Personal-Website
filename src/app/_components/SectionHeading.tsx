export type SectionHeadingProps = React.PropsWithChildren<{
  title: string;
  headline: string;
  body?: string;
}>;

const SectionHeading: React.FC<SectionHeadingProps> = ({
  title,
  headline,
  body,
  ...props
}) => {
  return (
    <div className="text-center max-w-3xl mx-auto" {...props}>
      <span className="text-primary font-semibold text-xl uppercase">
        {headline}
      </span>
      <h2 className="text-4xl font-bold">{title}</h2>
      {body && <p className="text-body mt-4">{body}</p>}
    </div>
  );
};

export default SectionHeading;
