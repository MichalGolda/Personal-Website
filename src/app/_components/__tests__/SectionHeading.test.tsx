import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import SectionHeading from "../SectionHeading";

describe("SectionHeading component", () => {
  it("renders without errors", () => {
    render(
      <SectionHeading
        title="test title"
        headline="test headline"
        body="test body"
      />
    );

    const title = screen.getByText("test title");

    expect(title).toBeInTheDocument();
    expect(title.tagName).toBe("H2");

    const headline = screen.getByText("test headline");

    expect(headline).toBeInTheDocument();
    expect(headline.tagName).toBe("SPAN");

    const body = screen.getByText("test body");

    expect(body).toBeInTheDocument();
    expect(body.tagName).toBe("P");
  });

  it("renders without body", () => {
    const { container } = render(
      <SectionHeading title="test title" headline="test headline" />
    );

    const sectionHeading = container.firstChild;

    expect(sectionHeading?.childNodes.length).toBe(2);
  });

  it("renders unchanged", () => {
    const { container } = render(
      <SectionHeading
        title="test title"
        headline="test headline"
        body="test body"
      />
    );

    expect(container).toMatchSnapshot();
  });
});
