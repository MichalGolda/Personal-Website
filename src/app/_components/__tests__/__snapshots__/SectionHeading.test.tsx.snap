// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeading component renders unchanged 1`] = `
<div>
  <div
    class="text-center max-w-3xl mx-auto"
  >
    <span
      class="text-primary font-semibold text-md md:text-lg lg:text-xl uppercase"
    >
      test headline
    </span>
    <h2
      class="text-2xl md:text-3xl lg:text-4xl text-secondary font-bold"
    >
      test title
    </h2>
    <p
      class="text-sm md:text-base text-body mt-4"
    >
      test body
    </p>
  </div>
</div>
`;
